src/NuGet.Core/NuGet.Packaging/Signing/Signatures/AuthorPrimarySignature.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Globalization;
+#if IS_DESKTOP
+using System.Security.Cryptography.Pkcs;
+using System.Security.Cryptography.X509Certificates;
+#endif
+using NuGet.Common;
+
+namespace NuGet.Packaging.Signing
+{
+    public sealed class AuthorPrimarySignature : PrimarySignature
+    {
+#if IS_DESKTOP
+
+        public AuthorPrimarySignature(SignedCms signedCms)
+            : base(signedCms, SignatureType.Author)
+        {
+        }
+
+        internal override SignatureVerificationStatus Verify(
```

Comment:
Can't we just pass a settings object instead of a bunch of flags?  It's easier to maintain over time.